untitled.js

```diff
@@ -883,19 +400,19 @@ function rollshock(charid, token, unipenalty) {
 		var ir = randomInteger(6);
 		shockroll = shockroll + ir
 		if (i > 0) {
-			shockstr = shockstr + " + " + ir;
+			shockstr +=  " + " + ir;
 
 		} else {
-			shockstr = shockstr + "[[" + ir;
+			shockstr += `${unipenalty}d6: ${ir}`;
 
 		}
 	}
 	end = myGet("COMBAT_ENDURANCE", charid,0);
 	if (shockroll > end) {
 		token.set("status_sleepy");
-		return "<br/>Shock Roll: " + shockstr + "]]<br/><h4>FAIL</h4><br/>";
+		return "<br/>Shock Roll: " + labelMaker(shockroll,shockstr) + "<br/><h4>FAIL</h4>";
```

Comment:
so just a thought that occurred here
Instead of dealing with HTML strings all over the place in the code, you could potentially construct an object. An Attack Outcome object. Which has attributes like "shock_roll" and "shock_result" and "shock_outcome". As well as "attack_roll"->"1d100"; "attack_EML"->"53"; "attack_result"->"51"; "attack_outcome"->"MS".
Then, after you construct the entire attack_outcome object, you do all the formatting at once in one place.
I think this should be a later refactoring.

Reply:
created #37